mm2src/coins/qrc20/swap.rs

```diff
@@ -957,11 +946,20 @@ fn check_if_contract_call_completed(receipt: &TxReceipt) -> Result<(), String> {
     match receipt.excepted {
         Some(ref ex) if ex != "None" && ex != "none" => {
             let msg = match receipt.excepted_message {
-                Some(ref m) => format!(": {}", m),
-                None => String::default(),
+                Some(ref m) if !m.is_empty() => format!(": {}", m),
+                _ => String::default(),
             };
             ERR!("Contract call failed with an error: {}{}", ex, msg)
         },
         _ => Ok(()),
     }
 }
+
+fn is_swap_event_log(event_topic: &str, log: &LogEntry) -> bool {
+    let mut topics = log.topics.iter();
+    match topics.next() {
+        // every swap event should have only one topic in log
+        Some(first_event) => first_event == event_topic && topics.next().is_none(),
```

Comment:
Just asking: the E/QRC20 event can be also triggered during a swap. I guess it will be in a different `LogEntry`? In what cases do we have several topics in 1 `LogEntry`?

Reply:
As far as I understand your question, E/QRC20 event like `Transfer` event will be in a different `LogEntry`. But one `LogEntry` may contain several topics for example `Transfer`: https://github.com/KomodoPlatform/atomicDEX-API/blob/qrc20-native-mode/mm2src/coins/qrc20.rs#L1169
This condition is used to check the swap contract address emitted expected logs.

Reply:
I see, thanks for the explanation!